clinica/iotools/bids_utils.py

```diff
@@ -450,12 +457,13 @@ def write_sessions_tsv(bids_dir, sessions_dict):
         bids_id = sp.split(os.sep)[-1]
 
         if bids_id in sessions_dict:
-            session_df = pd.DataFrame(
-                sessions_dict[bids_id]["M00"],
-                index=[
-                    "i",
-                ],
-            )
+            columns = list(list(sessions_dict.values())[0].values())[0].keys()
+            session_df = pd.DataFrame(columns=columns)
+            it = 0
+            for k in sessions_dict[bids_id].keys():
+                session_dict = sessions_dict[bids_id][k]
+                session_df_temp = pd.DataFrame(session_dict, index=[k, ])
+                session_df = pd.concat([session_df, session_df_temp])
```

Comment:
Why do you need to build a DataFrame by successive concatenation? Can't you do it in one go, which is faster and more idiomatic?

Reply:
I'm not really a Python programmer: can you make me a code proposal to understand?